src/Monitors/ButtonMonitor.cpp

```diff
@@ -7,10 +7,37 @@ ButtonMonitor::ButtonMonitor(unsigned int offset)
 
 void ButtonMonitor::execute()
 {
-    sfr::button::pressed = digitalRead(constants::button::button_pin);
+    float val = digitalRead(constants::button::button_pin);
+    boolean possible_uncovered = false;
+
+    sfr::button::button_pressed->set_value(val);
+
+    // button is recognized as not pressed
+    if (val == 0) {
+
+        for (int mission_mode : sfr::mission::mode_history) {
+            // checks if mission mode has never reached burn
+            if (mission_mode >= sfr::mission::mandatoryBurns->get_id()) {
+                possible_uncovered = true;
+            }
```

Comment:
Since we now do the same check for both the photoresistor and the button, does it make sense just to have one boolean that is tripped once we have reached mandatory burns? I am concerned that if our mode history is large, this could cause task overrun issues.